scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
+
+        # Trying to create $dest by all means
+
+        # For case user is root, or has access to create $cmd. Else -> do 'then' section
+        if sh -c "$cmd" > /dev/null 2>&1 ; then
+            print "Creating $dest directory by running $cmd using $USER rights"
+        else
+            if which sudo > /dev/null 2>&1 || type sudo > /dev/null 2>&1 ; then
+                print "Creating $dest directory by running $cmd using sudo:"
+                if sudo -n true 2> /dev/null ; then
+                    print "Checked: sudo access does not require password."
+                else
+                    print "Checked: sudo access requires to enter password:"
+                fi
+                # Using sudo to create directory
+                if ! sudo sh -c "$cmd" && print "Exiting priviliged mode." ; then
+                    >&2 notice "
+
+    Password don't match or user $USER not in sudoers.
+    "
+                fi
+            else
+                print "'sudo' seems not installed"
             fi
-            added=1
-            break
+
         fi
-    done
 
-fi
+        # If dest still not created (with user or sudo) - use su to access root
+        if [ ! -d "$dest" ] ; then
+            print "Trying to get privileges using su and root password"
+            if ! su root sh -c "$cmd" && print "Exiting priviliged mode."; then
+                exit 1 # Look into the trap
+            fi
+        fi
+    }
+
+    createStore() {
+        trap createStoreRevert 1
+        createStoreRevert() {
+            error "
+
+    Script tried to create $dest/store , but it already exists.
+    Something is wrong with the process.
+    Please gather information of your setup conditions,
+    search does BUG exists make further actions.
+    "
+        }
+        mkdir -p "$dest"/store
+    }
+
+    copyNix() {
+        printf "copying Nix to %s..." "${dest}/store"
+
+        for i in $(cd "$self/store" >/dev/null && echo ./*); do
+            printf "." >&2
+            i_tmp="$dest/store/$i.$$"
+            if [ -e "$i_tmp" ]; then
+                rm -rf "$i_tmp"
+            fi
+            if ! [ -e "$dest/store/$i" ]; then   # If $dest/store/directory exists
+                cp -Rp "$self/store/$i" "$i_tmp" # Copy $self/store/directory to $dest/store/directory.PID
+                chmod -R a-w "$i_tmp"            # Remove write permissions in the whole $dest/store/directory.PID
+                chmod +w "$i_tmp"                # Add write permission to $dest/store/directory.PID
+                mv "$i_tmp" "$dest/store/$i"     # Place $dest/store/directory.PID as $dest/store/directory/directory.PID
```

Comment:
`Place $dest/store/directory.PID as $dest/store/directory`

Not sure you should be commenting all that code mind you

Reply:
Yeah, if anything I'd be commenting the less obvious things like `cp -p` (or expanding out the parameters), but there's no need IMO to comment every line.

Reply:
I've done that to try to understand what is happens here.
I've done it to return to it further, when I going to understand what happens here and why and how it works after.